src/Apps/Settings/Routes/Purchases/Components/SettingsPurchasesRow.tsx

```diff
@@ -85,14 +96,61 @@ const getPaymentMethodText = (
   }
 }
 
+const getOrderActionButton = (
+  displayState: CommerceBuyerOrderOfferStateEnum,
+  orderId: string
+) => {
+  switch (displayState) {
+    case "PAYMENT_FAILED":
+      return (
+        <Button
+          // @ts-ignore
+          as={RouterLink}
+          to={`/orders/${orderId}/payment/new`}
+          variant="primaryBlack"
```

Comment:
Is there a rule about limiting the number of primary buttons that appear on a page?